### PR TITLE
RL-update-v2.2.0

### PR DESCRIPTION
**Pull request by ivukovicRL**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
- [ ] Limit your pull request to one application and submit multiple pull requests if necessary.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by mjiyani-crest** [comment]
Hey @ivukovicRL, we will take care of these Static tests failures. If your other PR(https://github.com/splunk-soar-connectors/reversinglabs-tiscale/pull/4) is consisting the same changes, then you can close that PR.

Thanks,
Mit

**Comment by ivukovicRL** [comment]
Hi @mjiyani-crest and thanks for the support. RL-update PR is deleted. We have a few questions regarding the PR.

There was an obsolete view **tiscale_view.py** which we deleted in this PR but it was still there causing errors during the checks (as if it was cached somewhere). So we added it again as an empty file in this commit https://github.com/splunk-soar-connectors/reversinglabs-tiscale/pull/5/commits/a0e11a458d6390e3c9fa1d15f5f631891c7d65b1 so the test could pass. Can you delete it for us? 

We also need to update our docs before the merge. Do you have some kind of a template for the docs? Also is the md enough or do we need to generate both html and md for the docs?

**Comment by mjiyani-crest** [comment]
Hey @ivukovicRL, thanks for reverting back.

> There was an obsolete view **tiscale_view.py** which we deleted in this PR but it was still there causing errors during the checks (as if it was cached somewhere). So we added it again as an empty file in this commit [a0e11a4](https://github.com/splunk-soar-connectors/reversinglabs-tiscale/commit/a0e11a458d6390e3c9fa1d15f5f631891c7d65b1) so the test could pass. Can you delete it for us?

Once you will finish your changes, we will start working on this PR and take care of it.

> We also need to update our docs before the merge. Do you have some kind of template for the docs? Also is the md enough or do we need to generate both HTML and md for the docs?

Yes, you have to create/update an HTML file only from now onwards. You can add all the details which are necessary for the user to take care of while using the app.

**Comment by dgopani-crest** [comment]
Hello @ivukovicRL, Are you done with the changes? 
Can we start further activities to release the app?

**Comment by ivukovicRL** [comment]
> Hello @ivukovicRL, Are you done with the changes? Can we start further activities to release the app?

Hi @dgopani-crest, thanks for asking. We are almost done. There is is still one more deprecated function in our code, i will let you know when we fix this. 

Also we will soon open new PRs on https://github.com/splunk-soar-connectors/reversinglabs-ticloud and https://github.com/splunk-soar-connectors/reversinglabs-a1000

**Comment by dgopani-crest** [comment]
> Hi @dgopani-crest, thanks for asking. We are almost done. There is still one more deprecated function in our code, I will let you know when we fix this.

Thanks for the quick response @ivukovicRL. Please let us know once all the changes are updated, we will start further release activities

**Comment by ivukovicRL** [comment]
@dgopani-crest Is there a way to auto-generate a readme out of a config file (for actions outputs)? Also, do you have a preferred tool to convert a markdown to html? Is there a documentation describing how the documentation works? It seems that the docs for the app here 
image
are a combination of the markdown file and a config file. Is that correct?

**Comment by dgopani-crest** [comment]
@ivukovicRL, the Documentation mentioned in the screenshot will be auto-generated by your readme.html and <app>.json file and looks like [this](https://my.phantom.us/5.1/docs/app_reference/phantom_reversinglabs). The above section of `Configuration Variables` is generated by readme.html and below that is generated from json file.

readme.md is also a combination of these 2 files, which will help users to know about an app on the GitHub repo. For README.md do not worry, we will take care of that and will create it from mentioned 2 files. So you can update all the required data in the readme.html & json file, it will automatically reflect on the Documentation page

Also, this MR shows "This branch has conflicts that must be resolved" can you please check?
Screenshot 2022-01-12 at 4 17 50 PM

**Comment by ivukovicRL** [comment]
@dgopani-crest It should be ok now.

**Comment by dgopani-crest** [comment]
> @dgopani-crest It should be ok now.

Thanks @ivukovicRL, we will start further activities and release the app

**Comment by ebhoir-crest** [comment]
Hi @ivukovicRL, I am unable to push the changes to branch 'reversinglabs:rl-update2' with below error 'Permission to reversinglabs/reversinglabs-tiscale-splunk-soar.git denied to ebhoir-crest.' Could you please provide me the push access rights.

**Comment by ivukovicRL** [comment]
Hi @ebhoir-crest you are added as a collaborator. Please let me know if you have any more issues.

File: tiscale_consts.py
Comment (pdesai-crest) [inline tiscale_consts.py]: Fix this license

**Review by pdesai-crest — APPROVED** [review]

File: tiscale_consts.py
Comment (ebhoir-crest) [reply tiscale_consts.py]: fixed the license

**Comment by ebhoir-crest** [comment]
Static test failures are added to expected failures list and MR has been raised for the same. Once the expected failure MR is merged it will resolve the static tests.

File: NOTICE
Comment (ricardo-phantom) [inline NOTICE]: 2022

Comment (ebhoir-crest) [reply NOTICE]: Thanks @ricardo-phantom , updated the year

**Review by ngaglani-crest — APPROVED** [review]

**Review by ricardo-phantom — APPROVED** [review]

File: reversinglabs_view.py
Comment (rgil-splunk) [inline reversinglabs_view.py]:
```suggestion
        for key, value in values.items():
```

File: tiscale_connector.py
Comment (rgil-splunk) [inline tiscale_connector.py]: can we either make this a `debug_print` or remove it? also, use string formatting instead of concatenation if it's kept

File: tiscale_connector.py
Comment (rgil-splunk) [inline tiscale_connector.py]:
```suggestion
                        "REST POST API to server failed {}".format(e),
```

File: tiscale_connector.py
Comment (rgil-splunk) [inline tiscale_connector.py]:
```suggestion
                if TISCALE_JSON_API_KEY in config:
```

File: tiscale_connector.py
Comment (rgil-splunk) [inline tiscale_connector.py]:
```suggestion
                if TISCALE_JSON_API_KEY in config:
```

File: tiscale_connector.py
Comment (rgil-splunk) [inline tiscale_connector.py]: can we make the user agent string a constant? we seem to reuse it a couple of times

File: tiscale_connector.py
Comment (rgil-splunk) [inline tiscale_connector.py]:
```suggestion
                        "REST GET API to server failed {} url: {}".format(e, url),
```

File: tiscale_connector.py
Comment (rgil-splunk) [inline tiscale_connector.py]:
```suggestion
            file_data = next(iter(files_array))
```
to avoid generating the whole list

File: tiscale_connector.py
Comment (rgil-splunk) [inline tiscale_connector.py]:
```suggestion
        if phantom.is_fail(ret_val):
```

File: tiscale_connector.py
Comment (rgil-splunk) [inline tiscale_connector.py]:
```suggestion
            file_data = next(iter(files_array))
```

File: tiscale_consts.py
Comment (rgil-splunk) [inline tiscale_consts.py]:
```suggestion
TISCALE_MSG_MAX_POLLS_REACHED = "Reached max polling attempts. " \
    "Please use the MD5 or Sha256 of the file as a parameter to <b>get report</b> to query the report status."
```
a multiline string would preserve the literal spacing in the printed string

**Review by rgil-splunk — APPROVED** [review]

**Comment by ebhoir-crest** [comment]
Hi @ivukovicRL can you please suggest for the points to be added in unreleased.md for current app changes

**Comment by ebhoir-crest** [comment]
Hi @ivukovicRL can you please suggest points for unreleased.md ?

**Comment by dgopani-crest** [comment]
Hi @ivukovicRL, currently we have added below one point for release notes.  we are seeing backward compatibility changes in the action parameters so other than this point do we need to add any other release notes point?

- Compatibility changes for Python 3 support 

Also, we have made a few best practices changes in the PR. Please have a look into it and let us know in case of any queries.

**Comment by ivukovicRL** [comment]
@dgopani-crest  We agree with the changes.

**Comment by ebhoir-crest** [comment]
Static test fails as `min_phantom_version` will be `5.0.0`

**Comment by ivukovicRL** [comment]
Hi @dgopani-crest is there anything else that we need to do before the release?

Edit: It is released, thanks!

**Comment by dgopani-crest** [comment]
> Hi @dgopani-crest is there anything else that we need to do before the release?

@ivukovicRL, the app is released and available on the [splunkbase](https://splunkbase.splunk.com/app/6069/) and [my.phantom.us](https://my.phantom.us/5.0/apps/)

**Pull request merged**
